src/libstore/local-store.cc

```diff
@@ -136,7 +136,9 @@ LocalStore::LocalStore(
     for (auto & perUserDir : {profilesDir + "/per-user", gcRootsDir + "/per-user"}) {
         createDirs(perUserDir);
         if (!readOnly) {
-            if (chmod(perUserDir.c_str(), 0755) == -1)
+            struct stat st;
+            stat(perUserDir.c_str(), &st);
```

Comment:
This ignores errors from `stat()`. You could use `nix::lstat()` in `file-system.hh` which throws an exception in case of an error.

Reply:
That's not available on the plain Path type. I just used a plain if.

Reply:
Not sure what you mean? We have
```
struct stat stat(const Path & path);
struct stat lstat(const Path & path);
```

Reply:
Not sure why I didn't get it to work but now it works.